libs/dal/src/repositories/base-repository.ts

```diff
@@ -81,7 +82,27 @@ export class BaseRepository<T_DBModel, T_MappedEntity, T_Enforcement = object> {
     }
   }
 
+  private calcExpireDate(modelName: string, data: FilterQuery<T_DBModel> & T_Enforcement) {
+    const now: number = Date.now();
+    switch (modelName) {
+      case 'Message':
+        if (data.channel === 'in_app') {
+          return addMonths(now, 6);
+        } else {
+          return addMonths(now, 1);
+        }
+      case 'Notification':
+        return addMonths(now, 1);
+      default:
+        return null;
+    }
+  }
+
   async create(data: FilterQuery<T_DBModel> & T_Enforcement): Promise<T_MappedEntity> {
+    const expireAt = this.calcExpireDate(this.MongooseModel.modelName, data);
+    if (expireAt) {
+      data = { ...data, expireAt };
```

Comment:
```suggestion
      data.expireAt = expireAt;
```

Is there a particular reason for the ...destructuring? 

I think it is not needed as we just can assign the needed property and save some memory :)